form-flow/src/main/kotlin/com/ritense/formflow/service/FormFlowService.kt

```diff
@@ -39,13 +40,20 @@ class FormFlowService(
     }
 
     fun open(formFlowInstanceId: FormFlowInstanceId) {
+        processExpressions(formFlowInstanceId, FormFlowStep::onOpen)
+    }
+
+    fun complete(formFlowInstanceId: FormFlowInstanceId) {
+        processExpressions(formFlowInstanceId, FormFlowStep::onComplete)
+    }
+
+    private fun processExpressions(formFlowInstanceId: FormFlowInstanceId, expressionist: (FormFlowStep)-> List<String>?) {
         val formFlowInstance = formFlowInstanceRepository.getById(formFlowInstanceId)
         val currentStep = formFlowInstance.getCurrentStep()
-
-        val expressionProcessor = expressionProcessorFactory.create(currentStep)
+        val expressionProcessor = expressionProcessorFactory.create(mapOf("step" to currentStep))
```

Comment:
Having a bit of trouble getting my point across, but I don't think we should directly add the current step to the context. There's a lot of information in there that is irrelevant to the context, and as a result I think is better off not in there altogether. Maybe instead a DTO that contains the relevant data to that particular step? So something like the submission data for each form flow step instance up until the current step?

Reply:
Agreed. I can add an 'empty' DTO there of which the contents should be refined in another story.

Reply:
Sounds good to me!

Reply:
I ended up just adding a basic variable map. Adding a whole conversion-layer that can be extended by implementation teams is way out of scope for this story (although desirable I guess)

Reply:
Makes sense. Extending this is out of scope for this feature for now anyway.